src/cli.rs

```diff
@@ -522,10 +522,19 @@ arg_enum! {
 }
 
 impl Pretty {
+    #[cfg(not(target_os = "windows"))]
     pub fn color(self) -> bool {
         matches!(self, Pretty::colors | Pretty::all)
     }
 
+    #[cfg(target_os = "windows")]
+    pub fn color(self) -> bool {
+        match self {
+            Pretty::colors | Pretty::all => ansi_term::enable_ansi_support().is_ok(),
```

Comment:
Hm, I envisioned `.color()` as a simple utility function without side effects. It might be better to put this in `Printer::new()` instead.

I'm also working on a version that uses `colorterm`. An advantage would be that it looks at `$TERM` (and `$NO_COLOR`, so we don't have to) and doesn't make changes that persist after the program is done. But I'm not sure that it does everything we want and it might end up complicating the code, so it may not be a good idea. This approach should be fine to merge regardless.

Reply:
`enable_ansi_support()` effect doesn't seem to persist between program runs
```
C:\Users\shara\Desktop\projects\xh>target\debug\xh.exe httpbin.org/json
# normal colored output (binary built from this PR that calls enable_ansi_support())

C:\Users\shara\Desktop\projects\xh>xh httpbin.org/json
# garbled output (v0.8.0)
```

However, I believe the flag `enable_ansi_support()` enables [is not supported in older Windows versions](https://stackoverflow.com/a/46044584/5915221) so we may migrate to https://github.com/BurntSushi/termcolor in the future.